Examples/runtimes/python/DynamoDBEncryption/src/keyring/raw_rsa_keyring_example.py

```diff
@@ -0,0 +1,245 @@
+    assert put_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+    # 8. Get the item back from our table using the same client.
+    #    The client will decrypt the item client-side using the Raw RSA keyring
+    #    and return the original item.
+    key_to_get = {"partition_key": {"S": "rawRsaKeyringItem"}, "sort_key": {"N": "0"}}
+
+    get_response = encrypted_ddb_client.get_item(TableName=ddb_table_name, Key=key_to_get)
+
+    # Demonstrate that GetItem succeeded and returned the decrypted item
+    assert get_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+    returned_item = get_response["Item"]
+    assert returned_item["sensitive_data"]["S"] == "encrypt and sign me!"
+
+
+def should_generate_new_rsa_key_pair() -> bool:
+    """
+    Check if we need to generate a new RSA key pair.
+
+    :return: True if we need to generate a new key pair, False otherwise
+    """
+    # Check if a key pair already exists
+    private_key_file = os.path.exists(EXAMPLE_RSA_PRIVATE_KEY_FILENAME)
+    public_key_file = os.path.exists(EXAMPLE_RSA_PUBLIC_KEY_FILENAME)
+
+    # If a key pair already exists: do not overwrite existing key pair
+    if private_key_file and public_key_file:
+        return False
+
+    # If only one file is present: throw exception
+    if private_key_file and not public_key_file:
+        raise RuntimeError(f"Missing public key file at {EXAMPLE_RSA_PUBLIC_KEY_FILENAME}")
+    if not private_key_file and public_key_file:
+        raise RuntimeError(f"Missing private key file at {EXAMPLE_RSA_PRIVATE_KEY_FILENAME}")
+
+    # If neither file is present, generate a new key pair
+    return True
+
+
+def generate_rsa_key_pair():
+    """Generate a new RSA key pair and save to PEM files."""
+    # Safety check: Validate neither file is present
+    if os.path.exists(EXAMPLE_RSA_PRIVATE_KEY_FILENAME) or os.path.exists(EXAMPLE_RSA_PUBLIC_KEY_FILENAME):
+        raise RuntimeError("generateRsaKeyPair will not overwrite existing PEM files")
+
+    # This code will generate a new RSA key pair for example use.
+    # The public and private key will be written to the files:
+    #  - public: EXAMPLE_RSA_PUBLIC_KEY_FILENAME
+    #  - private: EXAMPLE_RSA_PRIVATE_KEY_FILENAME
+    # In practice, you should not generate this in your code, and should instead
+    # retrieve this key from a secure key management system (e.g. HSM)
+    # This key is created here for example purposes only.
+    private_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+
+    # Write private key PEM file
+    private_key_pem = private_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+
+    try:
+        with open(EXAMPLE_RSA_PRIVATE_KEY_FILENAME, "wb") as f:
+            f.write(private_key_pem)
+    except IOError as e:
+        raise RuntimeError("IOError while writing private key PEM") from e
```

Comment:
nit; ditto for other files. Basically prefer specific errors over generic RuntimeError (but this really isn't that important)

```suggestion
        raise OSError("IOError while writing private key PEM") from e
```